src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/UnsignedAssertionResponseHandlerTest.java

```diff
@@ -0,0 +1,244 @@
+package unit.uk.gov.ida.verifyserviceprovider.services;
+
+import com.google.common.collect.ImmutableSet;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.opensaml.core.xml.util.XMLObjectSupport;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.opensaml.saml.saml2.core.Attribute;
+import org.opensaml.saml.saml2.core.EncryptedAssertion;
+import org.opensaml.saml.saml2.core.Response;
+import org.opensaml.saml.saml2.encryption.Decrypter;
+import org.opensaml.saml.saml2.metadata.SPSSODescriptor;
+import org.opensaml.xmlsec.encryption.support.EncryptionConstants;
+import uk.gov.ida.saml.core.IdaConstants;
+import uk.gov.ida.saml.core.IdaSamlBootstrap;
+import uk.gov.ida.saml.core.extensions.eidas.CountrySamlResponse;
+import uk.gov.ida.saml.core.extensions.eidas.EncryptedAssertionKeys;
+import uk.gov.ida.saml.core.extensions.eidas.impl.CountrySamlResponseBuilder;
+import uk.gov.ida.saml.core.extensions.eidas.impl.EncryptedAssertionKeysBuilder;
+import uk.gov.ida.saml.core.validation.SamlResponseValidationException;
+import uk.gov.ida.saml.deserializers.StringToOpenSamlObjectTransformer;
+import uk.gov.ida.saml.security.EidasValidatorFactory;
+import uk.gov.ida.saml.security.SamlAssertionsSignatureValidator;
+import uk.gov.ida.saml.security.SecretKeyDecryptorFactory;
+import uk.gov.ida.saml.security.exception.SamlFailedToDecryptException;
+import uk.gov.ida.saml.security.validators.ValidatedAssertions;
+import uk.gov.ida.saml.security.validators.ValidatedResponse;
+import uk.gov.ida.saml.security.validators.encryptedelementtype.EncryptionAlgorithmValidator;
+import uk.gov.ida.verifyserviceprovider.services.UnsignedAssertionsResponseHandler;
+import uk.gov.ida.verifyserviceprovider.validators.InstantValidator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.ida.saml.core.test.builders.AssertionBuilder.anAssertion;
+import static uk.gov.ida.saml.core.test.builders.AssertionBuilder.anEidasAssertion;
+import static uk.gov.ida.saml.core.test.builders.AttributeStatementBuilder.anAttributeStatement;
+import static uk.gov.ida.saml.core.test.builders.ResponseBuilder.aResponse;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UnsignedAssertionResponseHandlerTest {
+
+    @Mock
+    EidasValidatorFactory eidasValidatorFactory;
```

Comment:
```suggestion
    private EidasValidatorFactory eidasValidatorFactory;
```

Reply:
Have added these all in one commit (well, smooshed into an existing commit to be accurate.)